span.go

```diff
@@ -444,43 +463,43 @@ func (s *Span) applySamplingDecision(decision SamplingDecision, lock bool) {
 }
 
 // Span can be written to if it is sampled or the sampling decision has not been finalized.
-func (s *Span) isWriteable() bool {
-	state := s.context.samplingState
+func (s SpanContext) isWriteable() bool {
```

Comment:
no objection to moving the methods onto span context struct, but let's move the actual code to the appropriate file as well